engine/streamengine/v2/engine.go

```diff
@@ -0,0 +1,103 @@
+package streamengine
+
+import (
+	"context"
+	"io"
+	"sync"
+
+	"github.com/arquivei/goduck"
+	"github.com/arquivei/goduck/gokithelper"
+	"github.com/go-kit/kit/endpoint"
+
+	"github.com/arquivei/foundationkit/errors"
+)
+
+// StreamEngine is an engine that processes a of messages from a stream, with
+// the order preserved.
+type StreamEngine struct {
+	streams   []goduck.Stream
+	nWorkers  int
+	handler   goduck.MessageHandler
+	workersWG *sync.WaitGroup
+
+	cancelFn       func()
+	processorError error
+}
+
+// NewFromEndpoint creates a StreamEngine from a go-kit endpoint
+func NewFromEndpoint(
+	e endpoint.Endpoint,
+	decoder goduck.EndpointMessageDecoder,
+	streams []goduck.Stream,
```

Comment:
We could change this to `...goduck.Strem` so programs with only one stream don't need to wrap it on a slice.